lifecycle-management/Ahk.Lifecycle.Management.DAL/Entities/LifecycleEvent.cs

```diff
@@ -1,11 +1,16 @@
+using Newtonsoft.Json;
+
 namespace Ahk.Lifecycle.Management.DAL.Entities
 {
     public abstract class LifecycleEvent
     {
-        public string Id { get; set; } = Guid.NewGuid().ToString();
-        public string Repository { get; set; } = null!;
-        public string Username { get; set; } = null!;
-        public abstract string EventType { get; set; }
-        public DateTime Timestamp { get; set; }
+        [JsonProperty("id")]
+        public abstract string Id { get; }
+        public abstract string Repository { get; }
+        public abstract string Username { get; }
+
+        [JsonProperty("$type")]
+        public abstract string Type { get; }
+        public abstract DateTime Timestamp { get; }
```

Comment:
Nem kell abstract legyen. Egyszerűen csak property-k, amiket a leszármazottak örökölnek. És akkor nem is kell megismételni őket a leszármazottakban.

Reply:
done, de így kell egy konstruktor az osztálynak

Reply:
így van, és pont ez a helyes delegálás az osztályok között